TKeazirian.HTTPServer/Router/Router.cs

```diff
@@ -0,0 +1,79 @@
+namespace TKeazirian.HTTPServer.Router;
+
+using Handler;
+using Response;
+using Request;
+
+public class Router
+{
+    private readonly RoutesConfig _routesConfig;
+
+    public Router(RoutesConfig routesConfig)
+    {
+        _routesConfig = routesConfig;
+    }
+
+    public Response Route(Request request)
+    {
+        if (RouteConfigContainsPath(request) && !IsHttpMethodAllowed(request))
+        {
+            return new NotImplementedResponse().BuildNotImplementedResponse();
+        }
+
+        if (RouteConfigContainsPath(request) && IsHttpMethodAllowed(request))
+        {
+            var handler = GetHandler(request);
+
+            if (IsHeadRequest(request) && HandlerContainsGetMethod(handler))
+            {
+                return new HeadResponse().BuildHeadResponse(handler, request);
+            }
+
+            if (IsOptionsResponse(request) && HandlerContainsGetMethod(handler))
+            {
+                return new OptionsResponse().BuildOptionsResponse(request, _routesConfig);
+            }
+
+            return handler.HandleResponse(request);
+        }
+
+        return new ResourceNotFoundHandler().HandleResponse(request);
+    }
```

Comment:
I like this pattern.  You would want to guard first then work, so I suggest changing the order here.
```csharp
    public Response Route(Request request)
    {
        if !RouteConfigContainsPath(request)
        {
            return new ResourceNotFoundHandler().HandleResponse(request);
        }

        if !IsHttpMethodAllowed(request)
        {
            return new NotImplementedResponse().BuildNotImplementedResponse();
        }

        var handler = GetHandler(request);

        if (IsHeadRequest(request) && HandlerContainsGetMethod(handler))
        {
            return new HeadResponse().BuildHeadResponse(handler, request);
        }

        if (IsOptionsResponse(request) && HandlerContainsGetMethod(handler))
        {
            return new OptionsResponse().BuildOptionsResponse(request, _routesConfig);
        }

        return handler.HandleResponse(request);
    }
```

Reply:
Okay I think I have implemented this change correctly!